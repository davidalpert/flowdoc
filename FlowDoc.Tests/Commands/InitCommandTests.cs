using System.IO;
using System.IO.Abstractions.TestingHelpers;
using System.Linq;
using ApprovalTests;
using FlowDoc.CLI.Commands;
using NUnit.Framework;
using ObjectApproval;

namespace FlowDoc.Tests.Commands
{
    [TestFixture]
    public class InitCommandTests
    {
        [Test]
        public void Execute()
        {
            var targetPath = Path.Combine("c:", "dev", "mydocs");

            var fileSystem = new MockFileSystem();

            var cmd = new InitCommand(fileSystem)
                      {
                          Options = new InitOptions()
                                    {
                                        TargetDirectory = targetPath
                                    }
                      };

            cmd.Execute();

            ObjectApprover.VerifyWithJson(fileSystem.AllFiles);
        }

        [Test]
        public void Execute_creates_a_table_of_contents()
        {
            var targetPath = Path.Combine("c:", "dev", "mydocs");
            var tocPath = Path.Combine(targetPath, "toc.md");

            var fileSystem = new MockFileSystem();

            var cmd = new InitCommand(fileSystem)
                      {
                          Options = new InitOptions()
                                    {
                                        TargetDirectory = targetPath
                                    }
                      };

            cmd.Execute();

            Assert.IsTrue(fileSystem.File.Exists(tocPath));
            Approvals.Verify(fileSystem.File.ReadAllText(tocPath));
        }

        [Test]
        public void NewMockFileSystem()
        {
            var fileSystem = new MockFileSystem();
            ObjectApprover.VerifyWithJson(fileSystem);
        }
    }
}
