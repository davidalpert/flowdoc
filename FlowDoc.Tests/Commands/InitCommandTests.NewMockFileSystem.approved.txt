{
    "File": {},
    "Directory": {},
    "FileInfo": {},
    "Path": {
        "AltDirectorySeparatorChar": {},
        "DirectorySeparatorChar": {},
        "InvalidPathChars": [
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {},
            {}
        ],
        "PathSeparator": {},
        "VolumeSeparatorChar": {}
    },
    "DirectoryInfo": {},
    "AllPaths": [],
    "AllFiles": [],
    "AllDirectories": []
}